addons/hr_timesheet/controllers/portal.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import OrderedDict
+from dateutil.relativedelta import relativedelta
+from operator import itemgetter
+
+from odoo import fields, http, _
+from odoo.http import request
+from odoo.tools import date_utils, groupby as groupbyelem
+from odoo.osv.expression import OR
+
+from odoo.addons.portal.controllers.portal import CustomerPortal, pager as portal_pager
+
+
+class CustomerPortal(CustomerPortal):
+
+    def _get_timesheet_domain(self):
```

Comment:
we should move this method to the model, and call the method `_timesheet_get_portal_domain()`